Enable entitlements by default

rjernst opened the pull request:
Entitlements are almost complete. This commit enables them by default, in preparation for 8.18/9.0.

rjernst committed:
Enable entitlements by default

Entitlements are almost complete. This commit enables them by default,
in preparation for 8.18/9.0.

elasticsearchmachine commented:
Pinging @elastic/es-core-infra (Team:Core/Infra)

mark-vieira reviewed (approved):
LGTM

prdoyle commented:
It needs [this](https://github.com/elastic/elasticsearch/blob/a8c9d2fc809de63cc38413d48dd2d48079c55427/distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/SystemJvmOptions.java#L187) specifically.

```
-Djdk.attach.allowAttachSelf=true
```

rjernst committed:
change default in cli too

rjernst committed:
Merge branch 'main' into entitlements/enable_by_default

prdoyle commented:
I think the [next problem](https://buildkite.com/elastic/elasticsearch-pull-request/builds/58495#0195244b-11f4-4b37-aff7-022d078185f2/95-11706) is the one @ldematte just fixed [here](https://github.com/elastic/elasticsearch/pull/123013)?

prdoyle committed:
Merge branch 'main' into entitlements/enable_by_default

rjernst committed:
add debugging

rjernst committed:
more debugging

Commit:
[CI] Auto commit changes from spotless

rjernst committed:
add leniency...

rjernst committed:
add pid file

Commit:
[CI] Auto commit changes from spotless

rjernst committed:
actually pass pid file

rjernst committed:
more debug

rjernst committed:
yet more debug

jdconrad committed:
prune extraneous paths to prevent incorrect binary search

jdconrad committed:
Update docs/changelog/123177.yaml

jdconrad committed:
Merge branch 'main' into prune

jdconrad committed:
Delete docs/changelog/123177.yaml

rjernst committed:
follow links

Commit:
[CI] Auto commit changes from spotless

rjernst committed:
trivially allow stuff from entitlements lib

jdconrad committed:
Merge branch 'main' into prune

rjernst committed:
don't do that

rjernst committed:
try again

Commit:
[CI] Auto commit changes from spotless

rjernst committed:
temp dir too

rjernst committed:
address comments

rjernst committed:
Merge branch 'prune' into entitlements/enable_by_default

rjernst committed:
less debug logging

rjernst committed:
log4j access

Commit:
[CI] Auto commit changes from spotless

rjernst committed:
guard linux paths

rjernst committed:
revert

rjernst committed:
Add platform property to files entitlement

Some file paths are OS specific. This commit adds a `platform` property
to each file in a files entitlement that can be used to limit that file
to a specific platform.

rjernst committed:
Merge commit '1e23a291d6203b171683e8773fa30a986605a388' into entitlements/enable_by_default

rjernst committed:
fix logic

rjernst committed:
fix logic

rjernst committed:
silly copy paste

rjernst committed:
silly copy paste

rjernst committed:
more entitlements

rjernst committed:
Add shared_repo relative dir for files entitlement

This commit adds the shared repo path as a relative base dir. However,
it does not make this available to policy files, only to server.

rjernst committed:
Merge branch 'main' into entitlements/enable_by_default

rjernst committed:
spotless

mosche committed:
platform specific named pipe for windows

mosche committed:
Merge branch 'entitlements/platform_files' into entitlements/enable_by_default

ldematte committed:
Workaround to always allow named pipes on windows

ldematte committed:
Skip absolute/relative path validation if platform is "any" (null)

mosche committed:
More robust checking of parent path on Windows

mosche committed:
Revert "Workaround to always allow named pipes on windows"

This reverts commit c81d28d3530efa391e7092b6aaf2b484e99614ff.

mosche committed:
Merge branch 'entitlements/windows_pipes_normalization' into entitlements/enable_by_default

ldematte committed:
Some missing entitlements preventing serverless to start

prdoyle committed:
Merge remote-tracking branch 'upstream/main' into enable-entitlements

rjernst committed:
fix absolute checks

ldematte committed:
Merge remote-tracking branch 'upstream/main' into entitlements/enable_by_default

ldematte committed:
Merge branch 'entitlements/enable_by_default' of https://github.com/rjernst/elasticsearch into entitlements/enable_by_default

ldematte committed:
Merge remote-tracking branch 'upstream/main' into entitlements/enable_by_default

ldematte committed:
Merge remote-tracking branch 'upstream/main' into entitlements/enable_by_default

ldematte committed:
Merge remote-tracking branch 'upstream/main' into entitlements/enable_by_default
